src/main/java/com/good/works/summer/project/service/TeamService.java

```diff
@@ -115,15 +116,32 @@ public void ifOrganizationHasMoreThanFiveTeamsInSameCity(Team team) throws TeamS
     }
 
 
-    public List<Team> filterTeamsByCategory(Category categoryTitle){
+    public List<Team> filterTeamsByCategory(Category categoryTitle) {
         List<Team> filteredTeamsList = teamRepository.findAll();
         filteredTeamsList = filteredTeamsList.stream()
-                .filter(team -> team.getIdeas().stream()
-                        .anyMatch(idea -> idea.getProject().getCategory().equals(categoryTitle)))
+                .filter(team -> team.getIdeas().stream().anyMatch(idea -> idea.getProject().getCategory().equals(categoryTitle)
+                        && idea.getProject().isApproved()))
+                .sorted((first, second) -> second.getId() - first.getId())
                 .collect(Collectors.toList());
         return filteredTeamsList;
     }
 
+    public void approveProjectByID(int projectId) {
+        List<Team> teamList = this.teamRepository.findAll();
 
 
+        /*teamList.stream()
+                .flatMap(team -> team.getIdeas().stream())
+                .filter(idea->idea.getId()== projectId)
+                .forEach(idea->idea.getProject().setApproved());
+
+        */
+        for(Team team : teamList){
+            for(Idea idea: team.getIdeas()){
+                if( idea.getProject().getId() == projectId){
```

Comment:
Some files formatted, others skipped. Format everything in a separate PR, away from new changes.